multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1261,6 +1261,7 @@ def _fit(
             monitor=task.validation_metric_name,
             patience=config.optimization.patience,
             mode=minmax_mode,
+            stopping_threshold=1 - 1e-6 if minmax_mode == MAX else 1e-6,
```

Comment:
If it's RMSE, the loss functions can actually go very small so 1E-6 may not be a good number.

We can extend METRIC_MODE_MAP https://github.com/autogluon/autogluon/blob/1386d0ecc5069c5837c048970afd64d086745b68/multimodal/src/autogluon/multimodal/constants.py#L88 and write a METRIC_STOPPING_THRESHOLD_MAP to discuss the stopping threshold case-by-case.

Reply:
```
from sklearn.metrics import mean_squared_error
mean_squared_error([1], [1-1e-6], squared=False) ## gives value 1.0000000000287557e-06
mean_squared_error([1], [1-1e-6], squared=True) ## gives value 1.0000000000575112e-12
```
If the rmse metric reaches 1e-6, the mse training loss would be around 1e-12, which probably can't provide enough gradient.

Reply:
I'm not sure whether 1E-6 for validation RMSE can be ignored for gradient. How about for us to not add the stopping threshold for RMSE for now since we haven't seen examples of RMSE hitting a perfect value. Usually, we should always allow the model to improve during the training process.

For ROC-AUC, will stopping_threshold=1.0 work?

In addition, we need to separately discuss the stopping threshold for DIRECT_LOSS, LOG_LOSS, CROSS_ENTROPY and other loss functions. So we may adopt the same logic as in https://github.com/autogluon/autogluon/blob/1386d0ecc5069c5837c048970afd64d086745b68/multimodal/src/autogluon/multimodal/constants.py#L88-L106

We can just create a `METRIC_STOPPING_THRESHOLD` variable.

Reply:
Okay, just noticed that you have tested for stopping_threshold = 1 and it won’t work for ROC_AUC. So 1 - 1E-6 might be good for ROC_AUC. Or 1 - 1E-9 to play safer?

Reply:
I find that 1e-7 works, but 1e-8 or smaller doesn't stop early. I choose 1e-6 to be safe.

Reply:
May be 1E-7 in case some methods are really strong?

Reply:
Changed it to 1e-7.